File: schemas/friendships.py
from pydantic import BaseModel
from datetime import datetime

class FriendshipBase(BaseModel):
    user_id: int

class FriendshipCreate(FriendshipBase):  # Inherit from FriendshipBase
    friend_id: int

class Friendship(FriendshipBase):
    id: int
    friend_id: int
    created_at: datetime

    class Config:
        from_attributes = True 

class FriendshipResponse(BaseModel):
    id: str
    user_id: str
    friend_id: str
    created_at: datetime

File: schemas/friendships.js
const mongoose = require('mongoose');

// FriendshipBase model
const friendshipBaseSchema = new mongoose.Schema({
  user_id: { type: Number, required: true },
});

// FriendshipCreate model (inherits from FriendshipBase)
const friendshipCreateSchema = new mongoose.Schema({
  user_id: { type: Number, required: true },
  friend_id: { type: Number, required: true },
});

// Friendship model
const friendshipSchema = new mongoose.Schema({
  user_id: { type: Number, required: true },
  friend_id: { type: Number, required: true },
  created_at: { type: Date, default: Date.now },
});

// FriendshipResponse model
const friendshipResponseSchema = new mongoose.Schema({
  id: { type: String, required: true },
  user_id: { type: String, required: true },
  friend_id: { type: String, required: true },
  created_at: { type: Date, default: Date.now },
});

// Create Mongoose models
const Friendship = mongoose.model('Friendship', friendshipSchema);
const FriendshipCreate = mongoose.model('FriendshipCreate', friendshipCreateSchema);
const FriendshipResponse = mongoose.model('FriendshipResponse', friendshipResponseSchema);

// Export models for use in other parts of the app
module.exports = { Friendship, FriendshipCreate, FriendshipResponse };
